Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,254 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+public class BPKSaveCardButton: UIButton {
+
+    public var checked: Bool {
```

Comment:
Should this button have the 'selected' trait? 
Or do we expect consumers to set the accessibilityLabel to "Saved to list" or "Save to list"

Reply:
We expect consumers to set the `accessibilityLabel` 
This property is added to the constructor.

```swift
 public init(
        accessibilityLabel: String,
        style: BPKCardButtonStyle = .default,
        size: BPKCardButtonSize = .default
    )
```

Reply:
Do we expect them to update the label based on button state? Because now it's only on init. 
Feel like we need something to indicate the button is 'on'. adding/removing a `.selected` accessibilityTrait could perform that function

Reply:
In the Save to List use case,  BFF backend return the button's A11YText. 
When the button status change (after the network request), the button's accessibilityLabel is updated. 

At the same time, we can also overrride UIAccessibilityTraits. I will need to check the voice over behavior after adding this 

```swift
public override class func accessibilityTraits() -> UIAccessibilityTraits {
        checked ? .selected : super.accessibilityTraits()
    }
```

Reply:
The traits are like a 'set' you can insert and remove.
When the checked state change you can insert/remove the .selected trait.

This override is not that helpful to achieve the goal

Reply:
I added this function in this commit. 8750cbbeb467536b2471ecb60657c85fdbb4203f

```swift
    private func updateAccessibilityTraits() {
        if checked {
            accessibilityTraits.insert(.selected)
        } else {
            accessibilityTraits.remove(.selected)
        }
    }
```